meta/src/main/scala/scalan/meta/ScalanCodegen.scala

```diff
@@ -97,43 +120,35 @@ trait ScalanCodegen extends ScalanParsers { ctx: EntityManagement =>
     def getDefaultOfBT(tc: STraitCall) =
       s"DefaultOf${tc.name}" + tc.tpeSExprs.opt(args => s"[${args.rep()}]")
 
-    def zeroSExpr(t: STpeExpr): String = t match {
-      case STpePrimitive(_, defaultValueString) => defaultValueString
-      case STraitCall(name, args)
-        if module.entityOps.tpeArgs.exists(a => a.name == name && a.isHighKind) =>
-           s"c$name.lift(${args.rep(a => s"e${a}")}).defaultRepValue"
-      case tc @ STraitCall(name, args) => {
-        val optBT = module.entityOps.optBaseType
-        val isBT = optBT.exists(bt => bt.name == name)
-        if (isBT) {
-          getDefaultOfBT(tc) + ".value"
-          //s"Default.defaultOf[$t]"
-        } else {
-          s"element[$t].defaultRepValue"
+    def zeroSExpr(entity: STraitDef, t: STpeExpr): String = {
+      def zerpSExpr( t: STpeExpr) = zeroSExpr(entity, t)
+      t match {
+        case STpePrimitive(_, defaultValueString) => defaultValueString
+        case STraitCall(name, args)
+          if entity.tpeArgs.exists(a => a.name == name && a.isHighKind) =>
+          s"c$name.lift(${args.rep(a => s"e${a}")}).defaultRepValue"
+        case tc @ STraitCall(name, args) => {
+          val optBT = entity.optBaseType
+          val isBT = optBT.exists(bt => bt.name == name)
+          if (isBT) {
+            getDefaultOfBT(tc) + ".value"
+            //s"Default.defaultOf[$t]"
+          } else {
+            s"element[$t].defaultRepValue"
+          }
         }
+        case STpeTuple(items) => pairify(items.map(zerpSExpr))
```

Comment:
Yes, but in your code you have `items.map(zerpSExpr)` rather than `items.map(zeroSExpr)` (and can't have `items.map(zeroSExpr)` since it takes wrong amount of arguments).